test/util/deterministicuuid/deterministicuuid.go

```diff
@@ -0,0 +1,53 @@
+package deterministicuuid
+
+import (
+	"github.com/gofrs/uuid"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+const (
+	_ uint8 = iota
+	CLUSTERS
+	ASYNCOPERATIONS
+	PORTAL
+	GATEWAY
+)
+
+type gen struct {
+	namespace uint8
+	counter   uint16
+}
+
+func New(namespace uint8) uuid.Generator {
+	return &gen{
+		namespace: namespace,
+	}
+}
+
+func (g *gen) NewV1() (uuid.UUID, error) {
+	panic("unimplemented")
+}
+func (g *gen) NewV3(ns uuid.UUID, name string) uuid.UUID {
+	panic("unimplemented")
+}
+func (g *gen) NewV5(ns uuid.UUID, name string) uuid.UUID {
+	panic("unimplemented")
+}
+func (g *gen) NewV6() (uuid.UUID, error) {
+	panic("unimplemented")
+}
+func (g *gen) NewV7(uuid.Precision) (uuid.UUID, error) {
+	panic("unimplemented")
+}
```

Comment:
We are not using any of these in the codebase right ?
Couldn't we have an interface that is called `v4Generator` with a `NewV4` which `gofrs/uuid` would implement instead? That would avoid having these functions that just panic, and thus avoid bad surprises

Reply:
I'm also a little confused by our use of `V4` here, which according to [the UUID docs](https://github.com/gofrs/uuid#uuid) says are generated randomly per [RFC-4122](https://datatracker.ietf.org/doc/html/rfc4122#section-4.4), instead of deterministically. It doesn't sound like we're following any RFC here, so can we call the function something like `NewDeterministic` or something to avoid confusion, or am I missing something?

Reply:
As per the description, I am suspecting this is used for testing purposes.
@hawkowl would you please rename to more testing friendly name please?

Reply:
but if we keep it this way, we cannot change the naming since it is needed to implement the gofr interface.